File: AoC16/src/test/java/aoc16/d03/Day03Test.java
package aoc16.d03;

import aoc16.common.IOUtil;
import org.junit.Test;

import java.util.ArrayList;

import static org.junit.Assert.assertEquals;

public class Day03Test {

    public static boolean isTriangle( int a, int b, int c ) {
        return ( ( a + b ) > c ) && ( ( a + c ) > b ) && ( ( b + c ) > a );
    }

    @Test
    public void solution() {
        final var input = IOUtil.input("d03");
        final var list1 = new ArrayList<Integer>();
        final var list2 = new ArrayList<Integer>();
        final var list3 = new ArrayList<Integer>();
        int answer1 = 0;
        for ( final var l : input ) {
            final var a = l.split( "\\s+" );
            final var a1 = Integer.parseInt(a[1].trim() );
            final var a2 = Integer.parseInt(a[2].trim() );
            final var a3 = Integer.parseInt(a[3].trim() );
            answer1 += isTriangle( a1, a2, a3 ) ? 1 : 0;
            list1.add(a1);
            list2.add(a2);
            list3.add(a3);
        }
        assertEquals("answer 1", 1050, answer1 );
        int answer2 = 0;
        for ( int i = 0; i < list1.size(); i += 3 ) {
            answer2 += isTriangle( list1.get(i), list1.get(i+1), list1.get(i+2) ) ? 1 : 0;
            answer2 += isTriangle( list2.get(i), list2.get(i+1), list2.get(i+2) ) ? 1 : 0;
            answer2 += isTriangle( list3.get(i), list3.get(i+1), list3.get(i+2) ) ? 1 : 0;
        }
        assertEquals("answer 2", 1921, answer2 );
    }
}

File: AoC25/kotlin/src/test/kotlin/aoc25/Day03Test.kt
package aoc25

import kotlin.test.Test
import kotlin.test.assertEquals

class Day03Test {

    fun largestVoltage(bank:String): Int {
        return "${bank[0]}${bank[1]}".toInt()
    }

    @Test
    fun largestTest() {
        assertEquals(98, largestVoltage("987654321111111"))
    }

    fun solution(case: String): Pair<Int, Int> {
        val list = IOUtil.input(case)
        val answer1 = list.sumOf { bank -> largestVoltage(bank) }
        return answer1 to 0
    }

    @Test
    fun test() {
        val answer = solution("03-1")
        assertEquals(357, answer.first)
        assertEquals(0, answer.second)
    }

    @Test
    fun solution() {
        val answer = solution("03")
        assertEquals(0, answer.first)
        assertEquals(0, answer.second)
    }
}